### ISSUE TITLE
checking attrs of tag

### ISSUE THREAD

**Issue by odr** [snoyberg/xml#60]
The most common stream-parser for xml-tags now is `Text.XML.Stream.Parse.tag` but it ignore attrs. 
Having checking function `Name -> [(Text, Text)] -> Maybe a` could be convenient.

**Comment by k0ral** [comment]
I'm not sure I understand your point: the `tag` function lets you choose how to parse attributes through the `(a -> AttrParser b)` argument. Isn't that what you're requesting ?

**Comment by odr** [comment]
No, we can't say by AttrParser that element is incorrect.
It is about these lines of code for `tag`
```
        Just (EventBeginElement name as) ->
            case checkName name of
```
if `checkName` return Nothing no data from pipe processed.

I ask to change this code to
```
        Just (EventBeginElement name as) ->
            case checkName name as of
```
with some renaming for backward API compatibility.


E.g. I want this code should work
```
 choose 
    [ tag' (\(n,as) -> lookup "v1" as) ap1 c1
    , tag' (\(n,as) -> lookup "v2" as) ap2 c2
    ]
```
Now I can check attr only when EventBeginElement already fed to Conduit.

**Comment by k0ral** [comment]
This looks like a reasonable change to me.
I'll implement it as soon as I can, unless you want to make a pull request yourself.

(I took the liberty of reopening this issue, assuming you closed it by mistake ?)

**Comment by odr** [comment]
Right, it was my mistake.

I can make a pull request. I suppose "tag" have to keep signature. How to
call a "more common tag"?



2015-08-13 10:06 GMT+03:00 k0ral <notifications@github.com>:

**Comment by k0ral** [comment]
Implemented in #93.
